plugins/plugin-nodejs-debugger/che-plugin-nodejs-debugger-server/src/main/java/org/eclipse/che/plugin/nodejsdbg/server/NodeJsDebugProcess.java

```diff
@@ -32,70 +30,64 @@
  *
  * @author Anatoliy Bazko
  */
 public class NodeJsDebugProcess implements NodeJsProcessObservable {
-    private static final Logger LOG        = LoggerFactory.getLogger(NodeJsDebugProcess.class);
-    private static final int    MAX_OUTPUT = 4096;
-
+    private static final Logger LOG            = LoggerFactory.getLogger(NodeJsDebugProcess.class);
     private static final String NODEJS_COMMAND = detectNodeJsCommand();
 
-    private final Process                  process;
-    private final String                   outputSeparator;
-    private final ScheduledExecutorService executor;
-    private final BufferedWriter           processWriter;
-
+    private final Process                     process;
+    private final String                      outputSeparator;
+    private final ScheduledExecutorService    executor;
+    private final BufferedWriter              processWriter;
     private final List<NodeJsProcessObserver> observers;
 
     private NodeJsDebugProcess(String outputSeparator, String... options) throws NodeJsDebuggerException {
         this.observers = new CopyOnWriteArrayList<>();
         this.outputSeparator = outputSeparator;
 
-        List<String> commands = new ArrayList<>(1 + options.length);
-        commands.add(NODEJS_COMMAND);
-        commands.addAll(Arrays.asList(options));
-
-        ProcessBuilder processBuilder = new ProcessBuilder(commands);
-        try {
-            process = processBuilder.start();
-        } catch (IOException e) {
-            throw new NodeJsDebuggerException("NodeJs process failed.", e);
-        }
-
+        process = initializeNodeJsDebugProcess(options);
         processWriter = new BufferedWriter(new OutputStreamWriter(process.getOutputStream()));
 
-        executor = Executors.newScheduledThreadPool(1, new ThreadFactoryBuilder().setNameFormat("nodejs-debugger-%d")
-                                                                                 .setUncaughtExceptionHandler(
-                                                                                         LoggingUncaughtExceptionHandler.getInstance())
-                                                                                 .setDaemon(true)
-                                                                                 .build());
-        executor.scheduleWithFixedDelay(new OutputReader(), 0, 100, TimeUnit.MILLISECONDS);
+        executor = Executors
+                .newScheduledThreadPool(1, new ThreadFactoryBuilder()
+                        .setNameFormat("nodejs-debugger-%d")
+                        .setUncaughtExceptionHandler(LoggingUncaughtExceptionHandler.getInstance())
+                        .setDaemon(true)
+                        .build());
+
+        OutputReader outputReader = new OutputReader(process, outputSeparator, this::notifyObservers);
+        executor.scheduleWithFixedDelay(outputReader, 0, 100, TimeUnit.MILLISECONDS);
```

Comment:
`OutputReader` is not connected to container lifecycle.
It is started and stopped on demand.